bin/ansible-connection

```diff
@@ -291,9 +291,14 @@ def main():
         sys.stderr.write(traceback.format_exc())
         sys.exit("FAIL: %s" % e)
 
+    # disable logging on stdout, to avoid spurious debug logs to
+    # be outputted on stdout.
+    C.DEFAULT_LOG_ONLY = True
```

Comment:
Debug log message [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/__init__.py#L395) triggered by invoking `connection_loader` plugin in `ansible-connection` at [L294](https://github.com/ansible/ansible/blob/devel/bin/ansible-connection#L294) is the root cause of the problem. This debug message is not log only.